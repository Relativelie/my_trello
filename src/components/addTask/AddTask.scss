@import "../../templates";


.addTaskContainer {
    display: flex;
    justify-content: space-between;
    align-items: center;

    & input {
        width: 200px;
        padding: 3px;
        font-size: $tasksTextSize;
        font-family: $fontText;
    }

    & button {
        padding: 5px;
        border: none;
        border-radius: 10px;
        text-shadow: 0 1px 5px rgb(255, 255, 255);
        font-size: $tasksTextSize;
        font-family: $fontText;
        background-color: rgb(157, 209, 255);
        cursor: pointer;
    
        &:hover {
            box-shadow: 0 1px 10px rgba(255, 255, 255, 0.606);
            background-color: rgb(80, 168, 246);
        }
    }
}
