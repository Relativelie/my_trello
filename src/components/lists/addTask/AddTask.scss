@import "../../../templates";


.taskOption__adding {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 10px;

    & .taskOption__adding__inputField {
        width: 200px;
        padding: 3px;
        font-size: $tasksTextSize;
        font-family: $fontText;
    }

    & .taskOption__adding__addBtn {
        padding: 5px;
        border: none;
        border-radius: 10px;
        text-shadow: 0 1px 5px rgb(255, 255, 255);
        font-size: $tasksTextSize;
        font-family: $fontText;
        background-color: rgb(157, 209, 255);
        cursor: pointer;
    
        &:hover {
            box-shadow: 0 1px 10px rgba(255, 255, 255, 0.606);
            background-color: rgb(80, 168, 246);
        }
    }
}
