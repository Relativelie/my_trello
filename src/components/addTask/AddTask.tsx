import { FC, useRef } from 'react';
import { useActions } from '../../hooks/useActions';

import './AddTask.scss';

interface Props {
    indexOfList: number
}

export const AddTask: FC<Props> = ({ indexOfList }) => {
    const { addTask } = useActions();
    const myRef = useRef<HTMLInputElement>(null);

    const taskAdding = (e: string) => {
        if ((e === 'click' || e === 'Enter') && myRef.current !== null) {
            addTask(myRef.current.value, indexOfList);
            myRef.current.value = '';
        }
    };

    return (
        <div className="addTaskContainer">
            <input
                type="text"
                maxLength={34}
                onKeyPress={(e) => { taskAdding(e.code); }}
                ref={myRef}
            />
            <button
                onClick={(e) => taskAdding(e.type)}
                type="button"
            >
                Add task
            </button>
        </div>
    );
};
