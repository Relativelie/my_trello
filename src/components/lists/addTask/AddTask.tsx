import { FC, useRef } from 'react';
import { useActions } from '../../../hooks/useActions';

import './AddTask.scss';

interface Props {
    indexOfList: number
}

export const AddTask: FC<Props> = ({ indexOfList }) => {
    const { addTask, addTaskArrayToNewList } = useActions();
    const myRef = useRef<HTMLInputElement>(null);

    const taskAdding = (e: string) => {
        if ((e === 'click' || e === 'Enter') && myRef.current !== null) {
            addTaskArrayToNewList(indexOfList);
            addTask(myRef.current.value, indexOfList);
            myRef.current.value = '';
        }
    };

    return (
        <div className="addTaskContainer">
            <input
                aria-label="input task name"
                type="text"
                maxLength={34}
                onKeyPress={(e) => taskAdding(e.key)}
                ref={myRef}
            />
            <button
                onClick={(e) => taskAdding(e.type)}
                type="button"
            >
                Add task
            </button>
        </div>
    );
};
