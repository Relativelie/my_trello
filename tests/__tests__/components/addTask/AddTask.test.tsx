import '@testing-library/jest-dom/extend-expect';
import { fireEvent, screen } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';
import { AddTask } from '../../../../src/components/lists/addTask/AddTask';
import { render } from '../../../testsSetup/test-utils';

describe('Add task component', () => {
    test('snapshot - Add task component', () => {
        const tree = render(<AddTask indexOfList={0} />);
        expect(tree).toMatchSnapshot();
    });

    test('Add task - input task name(with enter press)', () => {
        render(<AddTask indexOfList={0} />);
        const inputEl = screen.getByLabelText(/input task name/i);
        userEvent.click(inputEl);
        userEvent.type(inputEl, 'some text for input 123!@#$%^&*()_');
        expect(inputEl).toHaveValue('some text for input 123!@#$%^&*()_');
        fireEvent.keyPress(inputEl, { key: 'Enter', keyCode: 13 });
        expect(inputEl).toHaveValue('');
    });

    test('Add task - input task name(with adding button click)', () => {
        render(<AddTask indexOfList={0} />);
        const inputEl = screen.getByLabelText(/input task name/i);
        userEvent.click(inputEl);
        userEvent.type(inputEl, 'some text for input 123!@#$%^&*()_');
        expect(inputEl).toHaveValue('some text for input 123!@#$%^&*()_');
        userEvent.click(screen.getByText(/add task/i));
        expect(inputEl).toHaveValue('');
    });
});
