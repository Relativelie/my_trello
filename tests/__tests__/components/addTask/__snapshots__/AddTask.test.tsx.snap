// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add task component snapshot - Add task component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="taskOption__adding"
      >
        <input
          aria-label="input task name"
          class="taskOption__adding__inputField"
          maxlength="34"
          type="text"
        />
        <button
          class="taskOption__adding__addBtn"
          type="button"
        >
          Add task
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="taskOption__adding"
    >
      <input
        aria-label="input task name"
        class="taskOption__adding__inputField"
        maxlength="34"
        type="text"
      />
      <button
        class="taskOption__adding__addBtn"
        type="button"
      >
        Add task
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
